ext/stringio/stringio.c

```diff
@@ -481,6 +492,9 @@ strio_set_string(VALUE self, VALUE string)
     rb_io_taint_check(self);
     ptr->flags &= ~FMODE_READWRITE;
     StringValue(string);
+    if (rb_str_chilled_p(string)) {
+        rb_str_modify(string);
+    }
     ptr->flags = OBJ_FROZEN(string) ? FMODE_READABLE : FMODE_READWRITE;
```

Comment:
Ditto.

```suggestion
    ptr->flags = OBJ_FROZEN(string) && !rb_str_chilled_p(string) ?
	FMODE_READABLE : FMODE_READWRITE;
```

Reply:
Thanks. The warnings are now emitted when the io is written to.